src/app/components/team/Languages/AddLanguageAction.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { commitMutation, createFragmentContainer, graphql } from 'react-relay/compat';
+import { Store } from 'react-relay/classic';
+import Button from '@material-ui/core/Button';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import TextField from '@material-ui/core/TextField';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { FormattedGlobalMessage } from '../../MappedMessage';
+import { FlashMessageSetterContext } from '../../FlashMessage';
+import GenericUnknownErrorMessage from '../../GenericUnknownErrorMessage';
+import { safelyParseJSON, getErrorMessageForRelayModernProblem } from '../../../helpers';
+import LanguageRegistry, { compareLanguages, languageLabel } from '../../../LanguageRegistry';
+
+function submitAddLanguage({
+  team,
+  languages,
+  onSuccess,
+  onFailure,
+}) {
+  commitMutation(Store, {
+    mutation: graphql`
+      mutation AddLanguageActionUpdateTeamMutation($input: UpdateTeamInput!) {
+        updateTeam(input: $input) {
+          team {
+            id
+            get_languages
+            rules_json_schema
+          }
+        }
+      }
+    `,
+    variables: {
+      input: {
+        id: team.id,
+        languages,
+      },
+    },
+    onError: onFailure,
+    onCompleted: ({ data, errors }) => {
+      if (errors) {
+        return onFailure(errors);
+      }
+      return onSuccess(data);
+    },
+  });
+}
+
+const AddLanguageAction = ({ team }) => {
+  const [dialogOpen, setDialogOpen] = React.useState(false);
+  const [isSaving, setIsSaving] = React.useState(false);
+  const [value, setValue] = React.useState(null);
+  const setFlashMessage = React.useContext(FlashMessageSetterContext);
+
+  const languages = safelyParseJSON(team.get_languages) || [];
+  const options = Object.keys(LanguageRegistry)
+    .filter(code => !languages.includes(code))
+    .sort((a, b) => compareLanguages(null, a, b));
+  const getOptionLabel = code => `${languageLabel(code)} (${code})`;
```

Comment:
"X (Y)" should be a `<FormattedMessage>`. Ref: https://www.mediawiki.org/wiki/Localisation#Symbols.2C_colons.2C_brackets.2C_etc._are_parts_of_messages

Reply:
Interesting! Looks like I'll need to use  `injectIntl` and `props.intl` on this one as the Autocomplete component expects a string and performs `.toLowerCase` on it

Reply:
Good catch. I think you can call it `renderOption` instead. (The default `renderOption()` implementation calls `getOptionLabel()`. I don't see why there are two methods in the first place....)